internal/characters/qiqi/qiqi.go

```diff
@@ -0,0 +1,176 @@
+package qiqi
+
+import (
+	"github.com/genshinsim/gcsim/pkg/character"
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+func init() {
+	core.RegisterCharFunc("qiqi", NewChar)
+}
+
+type char struct {
+	*character.Tmpl
+	talismanExpiry    []int
+	talismanICDExpiry []int
+	c4ICDExpiry       int
+	skillLastUsed     int
+}
+
+// TODO: Not implemented - C6 (revival mechanic, not suitable for sim)
+// C4 - Enemy Atk reduction, not useful in this sim version
+func NewChar(s *core.Core, p core.CharacterProfile) (core.Character, error) {
+	c := char{}
+	t, err := character.NewTemplateChar(s, p)
+	if err != nil {
+		return nil, err
+	}
+	c.Tmpl = t
+	c.Energy = 80
+	c.EnergyMax = 80
+	c.Weapon.Class = core.WeaponClassSword
+	c.NormalHitNum = 5
+	c.BurstCon = 3
+	c.SkillCon = 5
+
+	c.talismanExpiry = make([]int, len(c.Core.Targets))
```

Comment:
This should be in the character's init call similar to this:
https://github.com/genshinsim/gcsim/blob/0a1394e66d5b675ad6ba6ebbbe0dacf2612c87df/internal/characters/raiden/raiden.go#L48

make sure to call the template's init function

this is just in case for whatever reason the targets have not yet been initialized

Reply:
Wait does this need to be fixed in the Tartaglia code too then?

https://github.com/genshinsim/gcsim/blob/0a1394e66d5b675ad6ba6ebbbe0dacf2612c87df/internal/characters/tartaglia/tartaglia.go#L45

Reply:
> 

ya.. i thought i commented on that one before

Reply:
Well it got merged as-is so I used that as a reference haha. I'll change this in a bugfix PR.